pandas/tests/frame/test_constructors.py

```diff
@@ -2460,6 +2467,18 @@ def test_construct_with_two_categoricalindex_series(self):
         )
         tm.assert_frame_equal(result, expected)
 
+    def test_from_M8_structured(self):
+        dates = [(datetime(2012, 9, 9, 0, 0), datetime(2012, 9, 8, 15, 10))]
+        arr = np.array(dates, dtype=[("Date", "M8[us]"), ("Forecasting", "M8[us]")])
+        df = DataFrame(arr)
+
+        assert df["Date"][0] == dates[0][0]
+        assert df["Forecasting"][0] == dates[0][1]
+
+        s = Series(arr["Date"])
```

Comment:
maybe add a TODO for spliting out the Series test or parameterising.

as an aside where will the Series and Frame tests go. does this need to move again?

Reply:
> as an aside where will the Series and Frame tests go

For tests like this that are mostly testing Frame with a little bit of Series test on the side, it'll probably stay mixed for the foreseeable future.  For tests where we can share/parametrize tests between Series/Frame, maybe tests.generic? (though the tests currently in that directory are not in great shape)